### PR TITLE
docs: Docs nav structure reorg follow-up

### PR DESCRIPTION
**Pull request by aaronsteers**
I did a pass to clean up titles and organization a little bit, following from https://github.com/meltano/sdk/pull/1353#issuecomment-1414272456.

Preview:

- https://meltano-sdk--1407.org.readthedocs.build/en/1407/

Notable changes:

Removed "SDK Implementation Details - " as a prefix from pages in this section:

image

Moved "Plugin Capabilities" to Reference:

image

Combined "Contributing" page and "Release Process" under common parent. Raised "Deprecation Timeline" up one level to be visible to non-contributors/non-maintainers.

image

### CODE REVIEW

File: docs/CONTRIBUTING.md
Comment (WillDaSilva) [inline docs/CONTRIBUTING.md:1]:
```suggestion
# Contributing Guide
```

File: docs/development/release_process.md
Comment (WillDaSilva) [inline docs/development/release_process.md:5]:
```suggestion
Releases are published to PyPI by a GitHub Actions workflow, triggered when a GitHub [Release](https://github.com/meltano/sdk/releases) is published.
```

File: docs/index.rst
Comment (WillDaSilva) [inline docs/index.rst:79]: Why remove this?

Comment (aaronsteers) [reply docs/index.rst:79]: Moved under "development" parent.

image

If we raise up, we should raise up "Release Process" also; otherwise the "Development" parent would only have a single parent.

Comment (aaronsteers) [reply docs/index.rst:79]: Optionally, we could just have "Contributor's Guide", "Release Process", and "Deprecation Timeline" as top-level items under the "SDK Development" section.

@edgarrmondragon - any objections to us moving them up to the top level?

Comment (aaronsteers) [reply docs/index.rst:79]: As noted in the description, I do think "Deprecation Timeline" should be visible to non-contributors, since that effects tap/target maintainers even if they are not themselves contributors.

Comment (WillDaSilva) [reply docs/index.rst:79]: Oh I see where it is now. I was just failing to see the new location.

I have a slight preference for moving it to the top-level under "SDK Development". There's only 2 items nested there after all, and the name they're nested under is identical to the category name above.

**Review by WillDaSilva — APPROVED** [review]
Looks good to me. Moving those 2 items out from "SDK Development" would be nice, but this PR is still good with or without that.

File: docs/index.rst
Comment (edgarrmondragon) [reply docs/index.rst:79]: > any objections to us moving these up to the top level?

@aaronsteers no objections, it makes sense :+1:

**Review by edgarrmondragon — APPROVED** [review]
This LGTM, regardless if those two pages are moved up.

**Comment by aaronsteers** [comment]
I noticed we removed the dedicated `typing` page when consolidating into `classes`. Because there was helpful docstring at the top of that module, and because I like having a clean discoverable top-level nav for this, I've reverted the removal of the other page. I added `:noindex:` to the restored page so that the new pages are still the authority for class lookups.

**Pull request merged**
